### PR TITLE
Switch to go 1.23

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Use go 1.23 to build this project.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -110,7 +110,7 @@ require (
 	github.com/dimchansky/utfbom v1.1.1 // indirect
 	github.com/docker/cli v24.0.7+incompatible // indirect
 	github.com/docker/distribution v2.8.3+incompatible // indirect
-	github.com/docker/docker v26.1.5+incompatible // indirect
+	github.com/docker/docker v24.0.9+incompatible // indirect
```
Comment (ameukam) [inline go.mod:113]: Any reason why we downgrade ?

Comment (saschagrunert) [reply go.mod:113]: No, fixed that.

**Review by cpanato — APPROVED** [review]

**Comment by saschagrunert** [comment]
Green CI, PTAL @kubernetes-sigs/release-engineering

**Pull request merged**